internal/service/managedobjectstorage/user_access_key.go

```diff
@@ -0,0 +1,214 @@
+	svc := meta.(*service.Service)
+
+	var serviceUUID, username, name string
+	if err := unmarshalID(d.Id(), &serviceUUID, &username, &name); err != nil {
+		return diag.FromErr(err)
+	}
+
+	accessKey, err := svc.GetManagedObjectStorageUserAccessKey(ctx, &request.GetManagedObjectStorageUserAccessKeyRequest{
+		ServiceUUID: serviceUUID,
+		Username:    username,
+		Name:        name,
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return setManagedObjectStorageUserAccessKeyData(d, accessKey)
+}
+
+func resourceManagedObjectStorageUserAccessKeyUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
+	if !d.HasChange("enabled") {
+		return nil
+	}
+
+	svc := meta.(*service.Service)
+
+	var serviceUUID, username, name string
+	if err := unmarshalID(d.Id(), &serviceUUID, &username, &name); err != nil {
+		return diag.FromErr(err)
+	}
+
+	req := &request.ModifyManagedObjectStorageUserAccessKeyRequest{
+		Enabled:     d.Get("enabled").(bool),
+		ServiceUUID: serviceUUID,
+		Username:    username,
+		Name:        name,
+	}
+
+	accessKey, err := svc.ModifyManagedObjectStorageUserAccessKey(ctx, req)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return setManagedObjectStorageUserAccessKeyData(d, accessKey)
+}
+
+func resourceManagedObjectStorageUserAccessKeyDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	svc := meta.(*service.Service)
+
+	var serviceUUID, username, name string
+	if err := unmarshalID(d.Id(), &serviceUUID, &username, &name); err != nil {
+		return diag.FromErr(err)
+	}
+
+	req := &request.DeleteManagedObjectStorageUserAccessKeyRequest{
+		ServiceUUID: serviceUUID,
+		Username:    username,
+		Name:        name,
+	}
+
+	if err := svc.DeleteManagedObjectStorageUserAccessKey(ctx, req); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func setManagedObjectStorageUserAccessKeyData(d *schema.ResourceData, accessKey *upcloud.ManagedObjectStorageUserAccessKey) (diags diag.Diagnostics) {
+	if err := d.Set("access_key_id", accessKey.AccessKeyId); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("created_at", accessKey.CreatedAt.String()); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("enabled", accessKey.Enabled); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("last_used_at", accessKey.LastUsedAt.String()); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("name", accessKey.Name); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("updated_at", accessKey.UpdatedAt.String()); err != nil {
+		return diag.FromErr(err)
+	}
+	if accessKey.SecretAccessKey != nil {
+		if err := d.Set("secret_access_key", accessKey.SecretAccessKey); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return diags
+}
+
+func marshalID(components ...string) string {
```

Comment:
Do/should we have common helper for this? I think some other components use similar combined key as well

Reply:
Added [a separate PR](https://github.com/UpCloudLtd/terraform-provider-upcloud/pull/420).